src/pairing.cpp

```diff
@@ -351,19 +354,42 @@ void pairing(vector<unique_ptr<RoughReservoir>> &upper_reservoirs,
           upper_reservoir->pour_point, lower_reservoir->pour_point, coslat);
 
       if(upper_reservoir->brownfield){
+        distance_sqd = INF;
```

Comment:
distance_sqd and dist_sqd are not very distinct - perhaps rename to min_distance_sqd?

Reply:
Fixed